plugin/niassa+pinery/src/main/java/ca/on/oicr/gsi/shesmu/pinery/RemotePineryIUSRepository.java

```diff
@@ -0,0 +1,244 @@
+								limsAttr(sp, "geo_library_type", badSetInRecord::add, false).orElse(""), //
+								limsAttr(sp, "geo_prep_kit", badSetInRecord::add, false).orElse(""), //
+								sp.getLastModified() == null ? Instant.EPOCH : sp.getLastModified().toInstant(), //
+								new Tuple(sp.getSampleProvenanceId(), sp.getVersion(),
+										properties.getOrDefault("provider", "unknown")), //
+								sp.getCreatedDate() == null ? Instant.EPOCH : sp.getCreatedDate().toInstant(), //
+								"sample");
+
+						if (badSetInRecord.isEmpty()) {
+							return result;
+						} else {
+							badSetInRecord.forEach(name -> badSetCounts.merge(name, 1, (a, b) -> a + b));
+							return null;
+						}
+					})//
+					.filter(Objects::nonNull);
+
+		}
+
+		public Stream<PineryIUSValue> stream() {
+			if (Duration.between(lastUpdated, Instant.now()).get(ChronoUnit.SECONDS) > 900) {
+				provider.ifPresent(provider -> {
+					try (AutoCloseable timer = fetchLatency.start()) {
+						final Map<String, Integer> badSetCounts = new TreeMap<>();
+						final Map<String, String> runDirectories = new HashMap<>();
+						final Set<String> completeRuns;
+						try (PineryClient client = new PineryClient(properties.get("url"), true)) {
+							completeRuns = client.getSequencerRun().all().stream()//
+									.filter(run -> run.getState().equals("Completed"))//
+									.map(RunDto::getName)//
+									.collect(Collectors.toSet());
+						}
+						cache = Stream.concat(lanes(provider, badSetCounts, runDirectories, completeRuns::contains), //
+								samples(provider, badSetCounts, runDirectories, completeRuns::contains))//
+								.collect(Collectors.toList());
+						count.labels(fileName().toString()).set(cache.size());
+						lastUpdated = Instant.now();
+						lastFetchTime.labels(fileName().toString()).setToCurrentTime();
+						badSetCounts.entrySet()
+								.forEach(e -> badSetMap
+										.labels(Stream.concat(Stream.of(fileName().toString()),
+												Stream.of(e.getKey().split(":"))).toArray(String[]::new))
+										.set(e.getValue()));
+					} catch (final Exception e) {
+						e.printStackTrace();
+						provenanceError.labels(fileName().toString()).inc();
+					}
+				});
+			}
+			return cache.stream();
+		}
+
+		@Override
+		public Optional<Integer> update(ObjectNode value) {
+			properties = RuntimeSupport.stream(value.fields())
+					.collect(Collectors.toMap(Entry::getKey, e -> e.getValue().asText()));
+			properties.put("file", fileName().toString());
+			provider = Optional.of(new PineryProvenanceProvider(properties));
+			return Optional.empty();
+		}
+	}
+
+	private static final Gauge badSetMap = Gauge
+			.build("shesmu_pinery_ius_bad_set",
+					"The number of provenace records with sets not containing exactly one item.")
+			.labelNames("target", "property", "reason").register();
+
+	private static final Gauge count = Gauge
+			.build("shesmu_pinery_ius_last_count", "The number of lanes and samples from Pinery.").labelNames("target")
```

Comment:
Is this intended to count the sum total of lanes + samples? Or samples-in-lanes (IUSes)?

Reply:
If it's not counting IUSes, can the label be changed?

Reply:
Sure. Just `pinery`. It's cleaner that way.